R/GGI_PPI_Integrator.R

```diff
@@ -0,0 +1,415 @@
+# predictDirectedNetworks.R
+
+# Purpose: To augment a system with genetic and physical interaction data in order to hypothesize regulatory relationshipd between components.
+# Version: 1.0
+# Date: 2019-03-21
+# Author: Nada Elnour
+# License: MIT
+
+# Input: An excel spreadsheet of the system parsed using the BCB420-2019-resources scripts.
+# Output: Graph hypotheses of possible regulatry networks in the system
+# Dependencies: tibble 2.0.1; biomaRt 2.38.0; xlsx 0.6.1; readxl 1.3.1; dplyr 0.8.0.1; ggplot2 3.1.0; biogridr 0.0.0.9000; visNetwork 2.0.5
+# ==============================================================================
+
+# SIDE EFFECTS:
+# This script imports biogridr which uses the deprecated function xml2::xml_find_one().
+
+# ====  PACKAGES  ==============================================================
+#' @import devtools
+if (requireNamespace("biogridr", quietly = TRUE)) {
+  devtools::install_github("NElnour/biogridr", force = TRUE)
+  library(biogridr)
+} else {
+  library(biogridr)
+}
+
+# Load all required packages.
+require(xlsx, quietly = TRUE)
+require(readxl, quietly = TRUE)
+require(dplyr, quietly = TRUE)
+require(biomaRt, quietly = TRUE)
+require(ggplot2, quietly = TRUE)
+require(biogridr, quietly = TRUE)
+require(visNetwork, quietly = TRUE)
+
+source('./R/fetchData.R', echo=TRUE)
+source('./R/fetchComponents.R', echo=TRUE)
+source('./R/SyDButils.R', echo=TRUE)
+# ==============================================================================
+#' Filter system by physically-interacting components and return their genetic interactions
+#'
+#' @param sysName A string specifying the full path to the excel sheet containing the system's components
+#' @param key a unique string key
+#' @param criterion A string, either "stringent" or "relaxed", specifying whether only GGI between physical interactors should be selected.
+#' @return A dataframe of system components and either their GGI between physical interactors should be selected (iff \code{criterion} == "stringent")
+#' or all GGI of physical interactors (iff \code{relaxed} == "stringent")
+#' @import xlsx
+#' @import readxl
+#' @importFrom dplyr filter
+#' @import biomaRt
+#' @import ggplot2
+#' @import biogridr
+#' @import utils
+#' @include SyDButils.R fetchComponents.R fetchData.R
+#' @importFrom stats complete.cases
+#' @examples
+#' myKey <- getKey("Nada Elnour", "nada.elnour@@mail.utoronto.ca", "SILGRESA")
+#' mySys <- getSysInteractions("SLIGR", key = myKey, criterion = "stringent")
+#'
+#' @export
+getSysInteractions <-
+  function(sysName,
+           key,
+           criterion = "stringent") {
+    if (is.null(sysName)) {
+      stop("System not provided.\n")
+    } else if (is.null(key)){
+      stop("Provide valid key for biogrid ratification")
+    }
+
+    myURL <-
```

Comment:
The string edges data could be fetched by the "fetchdata" function, and the data itself is downloaded automatically whenever someone installs the package (so there is no need to download it again)